dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegretionTest.java

```diff
@@ -0,0 +1,418 @@
+import com.dotcms.datagen.*;
+import com.dotcms.util.CollectionsUtils;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Permission;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.portlets.htmlpageasset.business.render.HTMLPageAssetRenderedAPIImpl;
+import com.dotmarketing.portlets.htmlpageasset.business.render.PageContext;
+import com.dotmarketing.portlets.htmlpageasset.business.render.page.HTMLPageAssetRenderedBuilder;
+import com.dotmarketing.portlets.htmlpageasset.business.render.page.PageView;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.WebKeys;
+import com.liferay.portal.model.User;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class HTMLPageAssetRenderedAPIImplIntegretionTest extends IntegrationTestBase {
+
+    private HttpServletRequest request;
+    private Host host;
+    private User user;
+    private HttpServletResponse response;
+    private HTMLPageAsset htmlPageAsset;
+    private HttpSession session;
+    private Role role;
+
+    @BeforeClass
+    public static void prepare () throws Exception {
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    private void init () throws DotDataException, DotSecurityException {
+        init(null);
+    }
+
+    private void init (final Host host) throws DotSecurityException, DotDataException {
+        final User systemUser = APILocator.systemUser();
+        request = mock(HttpServletRequest.class);
+
+        session = mock(HttpSession.class);
+        when(request.getSession()).thenReturn(session);
+
+        response = mock(HttpServletResponse.class);
+
+        role = new RoleDataGen().nextPersisted();
+        user = new UserDataGen().roles(role).nextPersisted();
+
+        APILocator.getRoleAPI().addRoleToUser(APILocator.getRoleAPI().loadBackEndUserRole(), user);
+        this.host = host == null ? createHost() : host;
+
+        final Template template = new TemplateDataGen().nextPersisted();
+
+        htmlPageAsset = createPage(systemUser, role, template);
+    }
+
+    private HTMLPageAsset createPage(User systemUser, Role role, Template template) throws DotDataException, DotSecurityException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'template' is not assigned and could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4718001)